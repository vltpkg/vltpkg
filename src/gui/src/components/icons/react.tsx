export const React = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      width="500"
      height="500"
      viewBox="0 0 500 500"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}>
      <g clip-path="url(#clip0_2967_809)">
        <path
          d="m250.504 284.862c270.046 284.862 285.889 269.029 285.889 249.499c285.889 229.968 270.046 214.136 250.504 214.136c230.961 214.136 215.119 229.968 215.119 249.499c215.119 269.029 230.961 284.862 250.504 284.862z"
          fill="currentColor"
        />
        <path
          d="m250.501 321.96c355.363 321.96 440.371 289.522 440.371 249.509c440.371 209.495 355.363 177.058 250.501 177.058c145.639 177.058 60.6318 209.495 60.6318 249.509c60.6318 289.522 145.639 321.96 250.501 321.96z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.724 285.725c240.155 376.482 310.768 433.837 345.442 413.83c380.116 393.823 365.721 304.031 313.29 213.274c260.859 122.517 190.247 65.1621 155.572 85.1689c120.898 105.176 135.293 194.968 187.724 285.725z"
          stroke="currentColor"
          stroke-width="18.913"
        />
        <path
          d="m187.726 213.277c135.295 304.034 120.9 393.826 155.574 413.833c190.248 433.84 260.861 376.485 313.292 285.728c365.723 194.971 380.118 105.179 345.443 85.172c310.769 65.1652 240.157 122.52 187.726 213.277z"
          stroke="currentColor"
          stroke-width="18.913"
        />
      </g>
      <defs>
        <clipPath id="clip0_2967_809">
          <rect
            width="397"
            height="353"
            fill="currentColor"
            transform="translate(52 73)"
          />
        </clipPath>
      </defs>
    </svg>
  )
}
