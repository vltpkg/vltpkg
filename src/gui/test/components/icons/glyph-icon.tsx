import { afterEach, expect, test } from 'vitest'
import { cleanup, render } from '@testing-library/react'
import html from 'diffable-html'
import {
  GlyphIcon,
  ICONS,
  isGlyphIcon,
} from '@/components/icons/glyph-icon.jsx'

expect.addSnapshotSerializer({
  serialize: v => html(v),
  test: () => true,
})

afterEach(() => {
  cleanup()
})

test('renders glyph icon with default props', () => {
  render(<GlyphIcon icon="node" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('renders glyph icon with custom size', () => {
  render(<GlyphIcon icon="node" size="lg" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('renders glyph icon with custom color', () => {
  render(<GlyphIcon icon="node" color="blue" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('renders glyph icon with custom className', () => {
  render(<GlyphIcon icon="node" className="custom-class" />)
  expect(window.document.body.innerHTML).toMatchSnapshot()
})

test('renders different icon types', () => {
  Object.keys(ICONS).forEach(icon => {
    render(<GlyphIcon icon={icon as keyof typeof ICONS} />)
    expect(window.document.body.innerHTML).toMatchSnapshot()
  })
})

test('isGlyphIcon type guard works correctly', () => {
  expect(isGlyphIcon('node')).toBe(true)
  expect(isGlyphIcon('npm')).toBe(true)
  expect(isGlyphIcon('invalid-icon')).toBe(false)
})
