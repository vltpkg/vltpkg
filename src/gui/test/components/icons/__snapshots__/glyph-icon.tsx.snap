// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders different icon types 1`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 2`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 3`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 4`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 5`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 6`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders different icon types 7`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>
<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;

exports[`renders glyph icon with custom className 1`] = `

<div>
  <span class="font-zed antialised text-sm text-current custom-class">
    
  </span>
</div>

`;

exports[`renders glyph icon with custom color 1`] = `

<div>
  <span class="font-zed antialised text-sm text-blue-600 dark:text-blue-500">
    
  </span>
</div>

`;

exports[`renders glyph icon with custom size 1`] = `

<div>
  <span class="font-zed antialised text-lg text-current">
    
  </span>
</div>

`;

exports[`renders glyph icon with default props 1`] = `

<div>
  <span class="font-zed antialised text-sm text-current">
    
  </span>
</div>

`;
